\documentclass[12pt]{article}
 \usepackage[hcentering,bindingoffset=20mm]{geometry}
 \usepackage{placeins}
 \usepackage[numbib]{tocbibind}
 \usepackage{rotating}
\usepackage[square,sort,comma,numbers]{natbib}
 \usepackage{graphicx}
 \usepackage{tabularx}
 \linespread{1.3}
 \usepackage{gensymb}
\usepackage{longtable}
 \usepackage{lscape}
 \usepackage{url}
 \addtolength{\textwidth}{2cm}
 \addtolength{\hoffset}{-1cm}
 
 
 \addtolength{\textheight}{2cm}
 \addtolength{\voffset}{-1cm}
 \setlength{\parindent}{0pt}
 
\title{Redefinition of the evolutionary relationships within the gonyaulacales - wrangling paralogy in transcriptomics. $^{1}$}
\author{Key words: }
\date{}

\begin{document}
\maketitle
\paragraph{}Anna Liza Kretzschmar$^{2}$\\
Climate Change Cluster (C3), University of Technology Sydney, Ultimo, 2007 NSW, Australia, anna.kretzschmar@uts.edu.au
\paragraph{}Aaron Darling \\
Ithree institute (i3), University of Technology Sydney, Ultimo, 2007 NSW, Australia
\paragraph{}Mathieu Fourment \\
Ithree institute (i3), University of Technology Sydney, Ultimo, 2007 NSW, Australia
\paragraph{}Shauna Murray\\ 
Climate Change Cluster (C3), University of Technology Sydney, Ultimo, 2007 NSW, Australia
\newpage
\section{Abstract}
\newpage

\section{Introduction}
Advancement in next generation sequencing has heralded a new era for investigating the evolutionary relationships between organisms. 
The main focus has been on model organisms to infer the genetic make up of similar taxa. 
Model organisms are usually characterised by a reference genome and established protocols for working with. 
In the eukaryotic world, these well established model organisms only cover a predicted xx \% of known taxa. 
With the reduction in sequencing cost, increasingly large transcriptomic datasets are available for non-model organisms without reference genomes. 
Working with these taxa can pose obstacles for evolutionary inference when they do not satisfy the model selection or processing assumptions commonly attributed to model organisms. Frequent issues arise from:\
1) Selection of paralogs. If genes with different evolutionary histories are selected, the gene tree will not reflect the history of either paralog and be nonsensical for species tree inference; 
2) Concatenation of genes. Can be statistically inconsistent estimator of the species tree due to incomplete lineage sorting and concatenation acting as imperfect estimator of species tree topology \cite{roch2015likelihood}; and 
3) Inference of model adequacy from bootstrap values. Kubatko et al. (2007) demonstrated high bootstrap support under maximum likelihood inference for incorrect species trees with concatenated gene sets as input \cite{kubatko2007inconsistency}. As high bootstrap values are often used as an indicator for robust species topology resolution, this fallacy is particularly problematic if the reader/operator is unfamiliar with the statistical phenomenon.\\
In this study we aim to build a bioinformatic pipeline that attempts to addresses the issues of paralogy when processing transcriptomic data from non-model organisms, utilises alternative methods to both concatenation and maximum likelihood inference. 
Specifically, the pipeline assembles RNA seq datasets, identifies and extracts single copy genes across input taxa with extensive paralogy, and runs Bayesian inference phylogenetics. 
Finally, to apply the pipeline to the historically difficult order Gonyaulacales (phylum: Dinoflagellata) which are notorious for the issues that the pipeline seeks to address.\\

%protists and environmental importance
Dinoflagellates represent an ancient lineage on the eukaryotic branch of life. 
They play a role in several important ecological processes, as members are represented in aquatic environments, where the cover a diverse niches such as symbionts, parasites and some taxa can cause harmful algal blooms through proliferation and/or neurotoxin production.
While the evolutionary relationship of most orders within the dinoflagellates has been estimated with reasonable certainty, one order has consistently escaped scrutiny - the gonyaulacales. 
Analyses for this order commonly combined concatenation and maximum likelihood approaches, which yielded long branch attraction, low confidence values and inconsistent taxon resolution. 
Improving upon the species the evolution of the gonyaulaceles has implications for environmental disease monitoring, as neurotoxin production is prevalent in this order. 
Specifically the ciguatoxins produced by some \emph{Gambierdiscus} spp. are of interest as the causative agent of ciguatera fish poisoning, a neglected tropical disease with high associated morbidity.%, which is predicted to increase in prevalence with the advent of climate change.


\newpage

\section{Materials and methods}
\subsection*{Culture conditions}
Cultures were isolated from locations as per Table ~\ref{tbl:strainTable} and clonalal cultures established by micropipetting single cells through sterile seawater. 

\FloatBarrier
\begin{table}
\caption{Culturing conditions for species processed for this study.}
\label{tbl:strainTable}
\begin{tabular}{ | p{3cm} | p{1.5cm} | p{1.5cm} | p{6cm} |}
\hline
\textbf{Species} & \textbf{Strain}& \textbf{Temp} & \textbf{Source location} \\
\hline
\textit{Coolia malayensis}&MAB&& \\
\hline
\textit{Gambierdiscus carpenteri}&UTSMER9A&17&Merimbula, Australia\\
\hline
\textit{Gambierdiscus lapillus}&HG4&27&Heron Island, Australia\\
\hline
\textit{Gambierdiscus polynesiensis}&CG15&27&Rarotonga, Cook Islands\\
\hline
\textit{Gambierdiscus} cf. \textbf{silvae}&HG5&27&Heron Island, Australia\\
\hline
\textit{Ostreopsis ovata}&HER27&&\\
\hline
\textit{Ostreopsis rhodesae}&HER26&&\\
\hline
\textit{Ostreopsis siamensis}&BH1&&\\
\hline
\textit{Thecadinium kofoidii}&THECA&18&\\
\hline
\end{tabular}
\end{table}
\FloatBarrier
 \subsection*{RNA isolation}
\textit{Coolia malayensis}, \textit{Ostreopsis ovata}, \textit{Ostreopsis rhodesae} and \textit{Ostreopsis siamensis} were processed as per Ref ARJUN.
\emph{Gambierdiscus} spp. and \emph{Thecadinium kofoidii} were harvested during late exponential growth phase by filtration onto 5 $\mu$m SMWP Millipore membrane filter (Merck, DE) and washed off with sterile seawater. 
Cells were pelleted via centrifugation (make of centrifuge) for 10 minutes at 350 rcf. 
Supernatant was decanted and 2ml of TRI Reagent (Sigma-Aldrich, subsidiary of Merck, DE) was added to the pellet and vortexed till dissolved. 
Samples were split in two and transferred to 1.5ml eppendorf tubes. 
Cellular thecae were ruptured by three rounds of freeze-thaw, with tubes transferred between liquid Nitrogen and 95 $^{\circ}$C. 
RNA was extracted as per protocol for TRI Reagent. 
RNA elute was purified with the RNeasy RNA clean up kit EXACT NAME as per protocol(Qiagen). 
DNA was digested with Turbe DNAse (Life technologies, subsidiary of Thermo Fischer scientific, AU). 
RNA was quantified with Nanodrop 2000 (XXX) and frozen at -80 $^{\circ}$C until sequencing.
 
\subsection*{Library preparation and Sequencing}
The quality of samples was assessed via Agilent 2100 Bioanalyzer. 
Paired-end sequencing was performed on NextSeq 500 High Output was conducted by Ramaciotti center (UNSW, AU) with 75bp insert size for \emph{G. lapillus} and \emph{G.} cf. \emph{silvae}; and 150bp inserts for \emph{G. carpenterii}, \emph{G. polynesiensis} and \emph{T.} cf. \emph{kofoidii}.


%\subsection*{Transcriptomes}
%\emph{Coolia malayensis}, \emph{Gambierdiscus carpenteri}, \emph{Gambierdiscus lapillus}, \emph{G. polynesiensis}, \emph{Gambierdiscus} cf. \emph{silvae} \emph{Ostreopsis ovata}, \emph{Ostreopsis rhodesae}, \emph{Ostreopsis siamensis} and \emph{Thecadinium} cf. \emph{kofoidii} were grown in F/10 medium at 25 - 27 degrees celcius, except for \emph{T.} cf. \emph{kofoidii} which was grown at 18 degrees celcius, and cells were harvested via centrifugation at 350 rcf for 10 minutes at late exponential phase. Trizole was directly added to cells, cells were opened by three cycles of freezing in liquid nitrogen and thaw at 95 degrees celcius, then RNA was extracted as per reccomendation for Trizole by the manufacturers. Quality of RNA was screened by Agilent Bioanalyzer and paired-end sequencing was performed on NextSeq 500 High Output by Ramaciotti, UNSW. \emph{G. lapillus} and \emph{G.} cf. \emph{silvae} sequencing libraries were created with 75bp insert size; \emph{G. carpenterii}, \emph{G. polynesiensis} and \emph{T.} cf. \emph{kofoidii} sequencing libraries were created with 150bp inserts.
\subsubsection*{Publically available transcriptome libraries}
\emph{Gambierdiscus excentricus} VGO790 transcriptome was downloaded from NCBI under accession ID  SRR3348983 \cite{kohli2017role}.
RNA seq libraries for all remaining transcriptomes were generated by, and downloaded from, the Marine Microbial Eukaryote Transcriptome Sequencing Project \citep{keeling2014marine}.

\subsection*{Pipeline}
The pipeline is packaged in the Nextflow language for streamlined transferal between high powered computing clusters. 
The workflow is separated into part 1 and part 2, and modules within the pipeline are written in the Python 2.7 and bash languages.
\subsubsection*{Pipeline part 1}
The input to pipeline 1 were the individual RNA sequencing library which is the processed through FastQC for quality assurance, sequences are trimmed with Trimmomatic and assembled with Trinity v2.4.0. Assemblies were then processed with BUSCO2 with the protist specific library.
The RNA libraries with 150bp inserts generated as part of this study were also subjected to Digital Normalization to pool identical transcripts before assembly.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           
\subsubsection*{Pipeline part 2}
The input to pipeline 2 was the output of BUSCO2, where all complete single copy genes were identified per transcriptome. 
Any genes that were present in at least 75 \% the transcriptomes were indexed, the corresponding contig extracted from the assemblies, aligned with hmmer3 and unaligned regions trimmed. 
Species evolutionary inference was based on Bayesian probability with the *BEAST2 model in BEAST2. 
The analysis was performed under the WAG substitution model \cite{whelan2001general} with a Gamma distribution for four rate categories. 
A random local clock was employed \cite{drummond2010bayesian}. 
Posterior distributions of parameters were approximated after 300,000,000 generations of MCMC runs with 4 heated and one cold chain, sampled ever 5,000 generations  with a burn in of 15\%. 
The inference was run four times to compare convergence parameters, then log and tree files were merged. 
Inference was run using BEAGLE \cite{ayres2011beagle} on the University of Technology's High-powered computing cluster GPU.

\subsection*{Assembly analysis}
Contigs from assemblies from pipeline part 1 were searched against NCBI's non-redundant (nr) database in a BLASTx search with and E-value cut off $\leq$ 10$^{-3}$ in Blast2GO \cite{conesa2005blast2go}. Mapping and annotation were performed in Blast2GO also, including gene ontology (GO) term retrieval and enzyme code determination to establish metabolic pathways though the retrieval database.
\subsection*{Inference run time comparison}
A subset of 14 taxa was selected for a comparison between CPU and GPU inference run time comparison. 
The transcriptome libraries were fed through the pipeline as desctribed, which was set to extract single copy genes present in all of the taxa. 
The resulting alignments were run on *BEAST either on the CPU or GPU.

\newpage
\section{Results}
\FloatBarrier
\begin{longtable}{  | p{3cm} |p{2cm} | p{2cm} | p{2cm} | p{2cm} | p{2cm} |}
\caption{Summary of transcriptome sequencing and assembly statistics.}\\
\hline
\label{tbl:asmstats}
\emph{Sequences:}&\emph{G. carpenteri}&\emph{G. lapillus}&\emph{G. polynesiensis}&\emph{G.} cf. \emph{silvae}&\emph{T. kofoidii}\\
\hline
 \multicolumn{6}{| c |}{Sequencing}\\
 \hline
\textbf{SRA accession}&&&&&\\
\hline
\textbf{Raw sequencing reads}&&&&&\\
\hline
 \multicolumn{6}{| c |}{Assembly}\\
 \hline
 \textbf{Contigs}&&&&&\\
\hline
\textbf{Average length (bp)}&&&&&\\
\hline
\textbf{Range of contigs length (bp)}&&&&&\\
\hline
\textbf{GC content}&&&&&\\
\hline
  \multicolumn{6}{| c |}{Blast2GO}\\
\hline
\textbf{with BLASTx hit(E$\leq$ 10$^{-3}$)}&&&&&\\
\hline
\textbf{with GO-terms}&&&&&\\
\hline
\textbf{with full annotations}&&&&&\\
\hline
\textbf{with Enzyme Codes}&&&&&\\
\hline
\textbf{mapped to KEGG pathways}&&&&&\\
\hline
\end{longtable}

\FloatBarrier
\begin{longtable}{  | p{3cm} |p{4.5cm} | p{2cm} | p{2cm} | p{3cm} |}
\caption{Transcriptomes used for study along with taxonomic placement at family level and source. Family level placement derived from algaebase. MMETSP abbreviation for marine Microbial eukaryotic transcriptome sequencing project, by Moore Foundation.}\\
\hline
\label{tbl:Transcriptomes}
\textbf{Family}&\textbf{Species}&\textbf{Strain}&\textbf{BUSCO single complete}&\textbf{Source}\\
\hline
 \multicolumn{5}{| c |}{Gonyaulacales transcriptomes}\\
    \hline
   Ceratiaceae&\emph{Ceratium fusus}&PA161109&81&MMETSP1074 \citep{keeling2014marine}\\
        \hline
  Crypthecodiniaceae&\emph{Crypthecodinium cohnii}&Seligo&98&MMETSP0326\_2 \citep{keeling2014marine}\\
        \hline
    &\emph{Alexandrium catenella}&OF101&74&MMETSP0790 \citep{keeling2014marine}\\
        \hline
    &\emph{Alexandrium monilatum}&JR08&74&MMETSP0093 \citep{keeling2014marine}\\
        \hline
&\emph{Pyrodinium bahamense}&pbaha01&89&MMETSP0796 \citep{keeling2014marine}\\
        \hline
Gonyaulacaceae&\emph{Coolia malayensis}&MAB&100&This study\\
\hline
&\emph{Gambierdiscus carpenteri}&UTSMER9A&83&This study\\
\hline
&\emph{Gambierdiscus excentricus}&VGO790&83&\\
        \hline
    &\emph{Gambierdiscus lapillus}&HG4&98&This study\\
        \hline
            &\emph{Gambierdiscus polynesiensis}&CG14 really&81&This study\\
        \hline
    &\emph{Gambierdiscus} cf. \emph{silvae}&HG5&87&This study\\
        \hline
    &\emph{Gonyaulax spinifera}&CCMP409&53&MMETSP1439 \citep{keeling2014marine}\\
        \hline
%    &\emph{Lingulodinium polyedra}&CCMP1738&MMETSP1032&209&\citep{keeling2014marine}\\
 %       \hline
     &\emph{Ostreopsis ovata}&HER27&99&This study\\
     \hline
     &\emph{Ostreopsis rhodesae}&HER26&98&This study\\
     \hline
     &\emph{Ostreopsis siamensis}&BH1&98&This study\\
     \hline     
Protoceratiaceae&\emph{Protoceratium reticulatum}&CCCM535=CCMP1889&72&MMETSP0228 \citep{keeling2014marine}\\
    \hline
 &\emph{Thecadinium kofoidii}&THECA&70&This study\\
 \hline
 \multicolumn{5}{| c |}{Outgroup transcriptomes}\\
 \hline
Dinophysiaceae&\emph{Dinophysis acimunata}&DAEP01&74&MMETSP0797 \citep{keeling2014marine}\\
        \hline
Dinophyceae incertae sedis&\emph{Azadinium spinosum}&3D9&81&MMETSP1036\_2 \citep{keeling2014marine}\\
        \hline
Gymnodiniales&\emph{Karenia brevis}&CCMP2229&85&MMETSP0030 \citep{keeling2014marine}\\
%    \hline
%    &\emph{Amphidinium massartii}&&&&\\
%        &\emph{Durinskia baltica}&&&&\\
    \hline
%    &\emph{Peridinium aciculferum}&&&&\\
\end{longtable}

Azadinium Order Dinophyceae incerta sedis according to algaebase, gonyaulacales from NCBI taxonomy

%\FloatBarrier 
%\begin{figure} 
%\includegraphics[scale=.3]{HG7-env.png} 
%\caption{Detection of \emph{G. lapillus} per spatial replicate at each sampling site, cell numbers as normalised to HG7 standard curve (Fig. ~\ref{fig:lapiStd}B). Spatial replicates coloured as per macroalgal substrate,\ where \emph{Chnoospora} sp. are green, \emph{Sargassum} sp. are blue, \emph{Padina} sp. are red and mixed macroalgal substrates are yellow (see table ~\ref{tbl:MacroalgaeTable}).} 
%\label{fig:envHG7}
%\end{figure} 
%\FloatBarrier

\subsection*{Phylogeny}

\newpage
\section{Discussion}
This study presents a pipeline designed to handle the identification of single copy genes in ordinarily difficult taxa, extracting and aligning single copy gene sequences and running phylogenetic inference. 
Dinoflagellates are renowned for their large genomes and paralogy, which is why historically the evolutionary inference for them has been shithouse. With the recent completion and public availability of the MMETSP database, transcriptomes to investigate the evolutionary relationship for marine protists has become possible. And several studies have looked at the evolutionary relationships. While most clades of the dinoflagellates have been relatively well worked out, the gonyaulacales have been recalcitrant. So these have been of particular interest, especially as many of the taxa produce polyketide toxins that cause seafood intoxications globally. However, now that the transcriptomes are available, it's crucial to develop robust and reproducible methods to analyse the data
\subsection*{Phylogeny}
- differences between ML and BI analyses incl tanglegram
- literature on concatenation problem, incl. BS misleading (Kubatko 07, Roche 15)
- assembly diffs if transrate pans out
- something about single copy genes for paralog riddled taxa

\subsection*{Diff to previous gonyaulacales phylogenies}

The gonyaulacales phylogenies pre-dating this study differ extensively from the results presented here. As established, this is likely due to the methodologies used. A further factor is the available data for evolutionary inference. Before the MMETSP project, studies would either utilise a samll number of taxons with in depth sequencing, or taxons to cover the proposed family structure for this order but utulise only a small number of genes for investigation. The MMETSP database has allowed protists to join many other organisms on the big data set stage. With the access to data no longer as the limiting factor for analyses, the focus must now fall on the methods used to explore this data.
Several studies have been oublished employing the MMETSP database content since the project has gone live, for both direct exploration of the availavble data as well as supplementing trnascriptomes to answer questions. Arguably, there is a dosconnect between bioinformaticians, computer scientists and biologists which is well showcased in this area. The tennets of reproducible bioinformatics is hinged on open access software, as well as documentation, a sound understanding of the methods employed and best practices vs. computational trade-offs in analyses. As the gonyaulacales have proven a recalcitrant order for which to infer evolution, with a number of genetic peculiarities which make nalyses particularly prone to errors, this is an excellent case study.
The phylogeny presented by Price t al. 2017 is based on a similar assumption as this study - that single copy genes are the key to circumventing the issue of paralogy when selecting genes of interest. However, the methodology for identifying and selecting these single copy genes is not included in the publication. Further, contact with the author established that there was no script or documented steps for the pipeline that the authors executed. The bootstrap values for the resulting phylogeny are remarkably well supported, however without transparency for the gene selection for which genes were included or how the sequences were manipulated prior to phylogenetic inference the bootstrap values could simply be a case of the xxx phenomenon described by Kubatko et al. REF.
In Jackounevceck et al.'s 2016 study, the gonyaulacales phylogeny was part of the greater analyses around the phylum ?????, using the MMETSP transcriptomes to infer a number of evolutionary trends such as thecate evolution XXXXXX. NEED TO LOOK THIS HSIT UP AGAIN

The dominant method for established species trees is based on SSU and LSU rDNA sequences, usually within the genus and a small number of outgroups. The prevailing problem with this approach is that that deeper branches bwetween genera or higher taxonomic points is not well explored. Interestingly, the phylogenetic resolution of two genera with the highest species coverage, \emph{Gambierdiscus} and \emph{Ostreopsis}, did not mimic the resolution established in the literature within either genus. This is likely due to the rDNA sequences targeted for the usual studies reflect the evolution of that particular gene region, rather than the species evolution itself. However the arrangement of sister species for both \emph{Gambierdiscus} and \emph{Ostreopsis} matched that of the ketosynthase module of polyketide synthase complexes, which are thought to be involved in polyketide toxin production (Unpublished A. Kretzschmar; pers. comm. A. Verma). 

The use of a concatenation approach for the gene regions examined, coupled with ML for the inference, is commonly used in the literature. COncatenation as an approach suffers from a number of issues. The individual genes' evolutionary history can be quite distinct from the overall species evolution, yet they are treated as sharing the same evolutionary history, along with the rate in which the genes differentiated. This leads to a number of pitfalls (for details please see XXX), such as long branch attraction and incomplete lineage sorting, both which appear evident in studies using these methods for the gonyaulacales phylogeny elucidation (REF).
The BI matrix approach was chosen for this study to directly address these pitfalls. The resulting phylogeny (Fig. ) shows a well resolved, well supported inference of the gonyaulacales evolution. By presenting the pipeline designed in this study, as well as an example of running *BEAST on the GPU to drastically reduce running time, we hope that the use of reproducible, open-access processing of large data-sets such as the MMETSP database becomes the standard in the HAB community. The demonstration of utilising the GPU node on the high powered computing cluster to drastically reduce run time should hopefully encourage research teams to utilise the superior method at minimised cost of run time. 

The phylogeny presented here differs from previously published ... MONA's rtaxsonomy one..

Run time in *BEAST was compared with a test data-set of 22 single copy genes from 14 taxa, processed via the pipeline as described. THe inference was run on CPU, as well as GPU through the BEAGLE library.
The run time for the test data-set on CPU was 18 days. The run time on the GPU was 2 days, while the topology and nodal support of the resulting phylogenies matched.

- commonly used/required databases for taxon classification by journals, eg algaebase, and how the families don't match phylo
- Hoppenrath 17 suggesting reclassification as family 'asymetricomorpha' based on morphology, discuss differences and similarities
- diff of phylo to species trees within genera, eg. Ostreopsis and Gamb SSU, usually based on rDNA ie. gene like rather than species evolution
%- Saldarriaga 04 \cite{saldarriaga2004molecular} SSU & LSU gene based. ERGH neighbourhood joining method. Table 2: Gonya res as per Fensome 93
- Murray 05 \cite{murray2005improving} rDNA and only some gonya but looks like it meshes
 -orr 12 \cite{orr2012naked} puts Crypthecodinium in the Peridiniales too (Fig4)
 - SHALCHIAN-TABRIZI 06 \cite{shalchian2006combined} alsp places Crythecodinium in the Peridiniales(Fig 3)

Good for discussing findings which are different to literature to date Arroyave '11 Phylogenetic relationships and the temporal context for the diversification
of African characins of the family Alestidae (Ostariophysi: Characiformes): Evidence from DNA sequence data
- Zhang 07 use SSU cox1 and cob but only one or two gonya
- bachvaroff '14 74 protein matrix for phylo. they used concatenation approach and booted duplicate copies by selecting longest and some other rationales. C, chonii wasn't well resolved for them. Only moderate support for gonyaulacpods and prorocentrales but support dropped when gblocks trimming was put into effect. A.tamarenseshowed entirely differently placed copies of gene suggesting gene duplication confounding the analysis.  ref Bachvaroff and Place, 2008; Bachvaroff et al.,2009; Shoguchi et al., 2013 for gene duplication
-bachvaroff '11 concatenation approach 17 rDNA genes. only one gonya.
-derelle '16 stramenopile phylo but use concatenation which behaves badly with recombination/incomplete lineage sorting.
-price '17 talk about method problem
- the big other paper that ignore Shauna's findings





\newpage
\section{Conclusion}
\newpage

\section{Acknowledgements}
\bibliographystyle{acm}
\bibliography{/home/nurgling/PhD/writing/review/references.bib}


\end{document}